torch/csrc/autograd/cpp_hook.cpp

```diff
@@ -48,5 +48,22 @@ variable_list CppFunctionTensorPreHook::operator()(
   return results;
 }
 
+// NOLINTNEXTLINE(modernize-pass-by-value)
+CppFunctionSingleTensorPreHook::CppFunctionSingleTensorPreHook(
+    std::function<at::TensorBase(const at::TensorBase&)> hook,
+    int value_idx)
+    : hook_(hook), value_idx_(value_idx) {}
```

Comment:
@soulitzer Why not just move the hook here? No reason to make an expensive and unnecessary copy?

Reply:
Good catch, I overlooked this one

Reply:
Fixed in my general overhaul of std::move in autograd PR